src/csv.pp.ml

```diff
@@ -753,25 +785,30 @@ let output_record oc = function
 let output_all oc t =
   IF_LWT(Lwt_list.iter_s, List.iter) (fun r -> output_record oc r) t
 
-let print ?separator ?backslash_escape ?excel_tricks ?quote_all t =
+let print ?separator ?backslash_escape ?excel_tricks ?quote_all IF_EIO(~stdout,) t =
+  IF_EIO(Eio.Buf_write.with_flow stdout @@ fun w ->,)
   let csv = to_channel ?separator ?backslash_escape
-              ?excel_tricks ?quote_all IF_LWT(Lwt_io.stdout, stdout) in
+              ?excel_tricks ?quote_all IF_LWT(Lwt_io.stdout, IF_EIO(w, stdout)) in
   output_all csv t;%lwt
-  IF_LWT(Lwt_io.flush Lwt_io.stdout, flush stdout)
+  IF_LWT(Lwt_io.flush Lwt_io.stdout, IF_EIO(Eio.Buf_write.flush w, flush stdout))
 
-IF_LWT(,
+IF_LWT_EIO(,
 let save_out ?separator ?backslash_escape ?excel_tricks ch t =
   let csv = to_channel ?separator ?backslash_escape ?excel_tricks ch in
   output_all csv t
 )
 
 let save ?separator ?backslash_escape ?excel_tricks ?quote_all fname t =
+  IF_EIO(
+    Eio.Path.with_open_out ~create:(`Or_truncate 0o666) fname @@ fun sink ->
```

Comment:
The Lwt version calls `Lwt_io.open_file` without the `~flags` argument. I looked at the `lwt_io.ml` to make sure that the Eio version behaves identically.